src/WiFi.cpp

```diff
@@ -723,4 +733,54 @@ void WiFiClass::refresh(void)
 	m2m_wifi_handle_events(NULL);
 }
 
+uint8_t WiFiClass::ping(const char* hostname, uint8_t ttl){
+	IPAddress ip;
+	if (hostByName(hostname, ip) > 0)
+		return ping(ip, ttl);
+	else
+		return WL_PING_ERROR;
+};
+
+uint8_t WiFiClass::ping(const String &hostname, uint8_t ttl){
+	return ping(hostname.c_str(), ttl);
+};
+
```

Comment:
Since ```WiFiClient``` only has the following:

```c
	virtual int connect(IPAddress ip, uint16_t port);
	virtual int connect(const char* host, uint16_t port);
```

I think we should drop ```uint8_t WiFiClass::ping(const String &hostname, uint8_t tty)``` for now.

We can look into adding it to both later on.